diff -ruN olsrd-0.6.1.orig/make/Makefile.linux olsrd-0.6.1/make/Makefile.linux
--- olsrd-0.6.1.orig/make/Makefile.linux	2011-05-07 13:07:54.000000000 +0200
+++ olsrd-0.6.1/make/Makefile.linux	2011-05-07 13:08:27.000000000 +0200
@@ -19,7 +19,7 @@
 ifdef OLSRD_PLUGIN
 GENERATE_PIC = true
 endif
-ifeq ($(filter-out mips mips64 alpha x86_64 parisc64 ia64, $(ARCH)),)
+ifeq ($(filter-out armv7l mips mips64 alpha x86_64 parisc64 ia64, $(ARCH)),)
 GENERATE_PIC = true
 endif
 
