diff --git a/Makefile b/Makefile
index a9d7fb6..a63f59e 100644
--- a/Makefile
+++ b/Makefile
@@ -54,6 +54,8 @@ LDFLAGS +=	-Wl,--out-implib=libolsrd.a
 LDFLAGS +=	-Wl,--export-all-symbols
 endif
 
+LDFLAGS += -Wl,-z,now -pie
+
 SWITCHDIR =	src/olsr_switch
 CFGDIR =	src/cfgparser
 include $(CFGDIR)/local.mk
diff --git a/Makefile.inc b/Makefile.inc
index aa4f5c1..c5fa7cb 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -150,6 +151,7 @@ endif
 LDFLAGS +=	-shared
 LDFLAGS +=	-Wl,-soname,$(PLUGIN_SONAME)
 LDFLAGS +=	-Wl,--version-script=version-script.txt 
+LDFLAGS +=	-Wl,-z,now
 else
 # c and ld flags for main
 ifeq ($(OS),win32)
